### PR TITLE
Add ability to output raw binary

### PR DESCRIPTION
**Pull request by mlarouche**
Add InstallRawStep to Zig build system that does a similar job to llvm-objcopy. To use it, do
```zig
exe.installRaw("kernel.bin");'
```
where exe is a LibExeObjStep

Part of #2826

See commit https://github.com/wendigojaeger/ZigGBA/commit/4f7ffdc1688d7b088bbb8a6e45258ae12e52f71a for an example of usage with the build system

### CODE REVIEW

**Comment by LemonBoy** [comment]
Isn't this better suited as a easy-to-use build step rather than as a compiler feature?
The whole implementation needs _no_ data internal to the compiler and can be safely implemented in userspace as a pre-made build step the user can pick up and add to their own build process if needed.

**Comment by andrewrk** [comment]
I agree with @LemonBoy here. A pure userland solution integrated with zig build would solve the real use case, and then later in the self hosted compiler it can take advantage of this code to integrate with the CLI. We can get away with stage1 not having CLI support for this. And we should get away with as minimal stage1 as possible to keep complexity under control.

**Comment by mlarouche** [comment]
> 
> 
> I agree with @LemonBoy here. A pure userland solution integrated with zig build would solve the real use case, and then later in the self hosted compiler it can take advantage of this code to integrate with the CLI. We can get away with stage1 not having CLI support for this. And we should get away with as minimal stage1 as possible to keep complexity under control.

I'll change my PR to be userland-only in the zigh build system

**Comment by daurnimator** [comment]
See also the accepted proposal https://github.com/ziglang/zig/issues/2279#issuecomment-571802541

**Comment by mlarouche** [comment]
Pushed a new version with only build system code

File: lib/std/elf.zig [hunk 1]
```diff
@@ -443,47 +443,72 @@ pub const Elf = struct {
...
+        } else {
+            for (elf.program_headers) |*elf_program| {
+                elf_program.p_type = @as(Elf64_Word, try in.readInt(Elf32_Word, elf.endian));
+                elf_program.p_offset = @as(Elf64_Off, try in.readInt(Elf32_Off, elf.endian));
+                elf_program.p_vaddr = @as(Elf64_Addr, try in.readInt(Elf32_Addr, elf.endian));
+                elf_program.p_paddr = @as(Elf64_Addr, try in.readInt(Elf32_Addr, elf.endian));
+                elf_program.p_filesz = @as(Elf64_Word, try in.readInt(Elf32_Word, elf.endian));
+                elf_program.p_memsz = @as(Elf64_Word, try in.readInt(Elf32_Word, elf.endian));
+                elf_program.p_flags = @as(Elf64_Word, try in.readInt(Elf32_Word, elf.endian));
+                elf_program.p_align = @as(Elf64_Word, try in.readInt(Elf32_Word, elf.endian));
```
Comment (andrewrk) [inline lib/std/elf.zig]: I believe all these `@as` are superfluous.

File: lib/std/elf.zig [hunk 2]
```diff
@@ -443,47 +443,72 @@ pub const Elf = struct {
...
-                elf_section.addr_align = @as(u64, try in.readInt(u32, elf.endian));
-                elf_section.ent_size = @as(u64, try in.readInt(u32, elf.endian));
+                elf_section.sh_flags = @as(u64, try in.readInt(u32, elf.endian));
+                elf_section.sh_addr = @as(u64, try in.readInt(u32, elf.endian));
+                elf_section.sh_offset = @as(u64, try in.readInt(u32, elf.endian));
+                elf_section.sh_size = @as(u64, try in.readInt(u32, elf.endian));
+                elf_section.sh_link = try in.readInt(u32, elf.endian);
+                elf_section.sh_info = try in.readInt(u32, elf.endian);
+                elf_section.sh_addralign = @as(u64, try in.readInt(u32, elf.endian));
+                elf_section.sh_entsize = @as(u64, try in.readInt(u32, elf.endian));
```
Comment (andrewrk) [inline lib/std/elf.zig]: I believe this TODO comment can be deleted and the `@as` can be removed

**Pull request merged**